app/components/navigation_header/index.tsx

```diff
@@ -60,46 +60,50 @@ const NavigationHeader = forwardRef<SearchRef, Props>(({
     const theme = useTheme();
     const styles = getStyleSheet(theme);
 
+    const [heightOffset, setHeightOffset] = React.useState(0);
     const {largeHeight, defaultHeight, headerOffset} = useHeaderHeight();
+
+    useEffect(() => {
+        if (lockValue) {
+            setHeightOffset(lockValue);
+        } else {
+            setHeightOffset(headerOffset);
+        }
+    }, [lockValue, headerOffset]);
```

Comment:
A state that only gets updated inside an effect is a code smell.

Probably this doesn't need to be a state, but a simple variable do the trick. Something like...

```
heightOffset = lockValue || headerOffset;
```

If we keep it in the effect, we are forcing a new render when `lockValue` or `headerOffset` changes when we don't need that.

Reply:
Good point, I was so busy converting code, I didn't think about how it could have been simpler.